src/Dfc.ProviderPortal.Courses/Services/CoursesService.cs

```diff
@@ -346,6 +320,37 @@ public async Task<HttpResponseMessage> ArchiveCourseRunsByUKPRN(int UKPRN)
             }
         }
 
+        public async Task<HttpResponseMessage> ArchivePendingBulkUploadCourseRunsByUKPRN(int UKPRN)
+        {
+            Throw.IfNull<int>(UKPRN, nameof(UKPRN));
+            Throw.IfLessThan(0, UKPRN, nameof(UKPRN));
+
+            try
+            {
+                using (var client = _cosmosDbHelper.GetClient())
+                {
+                    var coursesToUpdate =  await _cosmosDbHelper.GetDocumentsByUKPRN(client, _settings.CoursesCollectionId, UKPRN);
+
+                    foreach (var course in coursesToUpdate.Where(
```

Comment:
This predicate would be better done in the Cosmos query

Reply:
Yeah I agree; I'm trying to make the minimal possible change to this test-less production code.
Do you want to suggest a patch, or discuss further in slack/video?